skfem/element/__init__.py

```diff
@@ -45,6 +45,7 @@
     ElementHex0: ElementQuad0,
     ElementHex1: ElementQuad1,
     ElementHex2: ElementQuad2,
+    ElementTetCCR: ElementTriCCR,
```

Comment:
If you want to include it here (for `InteriorBasis.trace` support), you should check that it works, e.g., by modifying https://github.com/kinnala/scikit-fem/blob/master/tests/test_basis.py#L193

If it doesn't work, then you should remove this line.

Reply:
Ok. Will check. Thanks

Reply:
Removing it for now. For the integration, should I add higher order quadrature points/weights?

Reply:
As for the higher order quadrature rules, I took these verbatim from [`ffc`](https://bitbucket.org/fenics-project/ffc/src/cb26c917c89bad1e5888836378cd3b395c077227/ffc/fiatinterface.py#lines-208), namely

```python3
from ffc.fiatinterface import create_quadrature
deg = 8 # 2, 3, 4 ...
points, weights = create_quadrature("tetrahedron", deg, "default")

```

Reply:
I'll verify the original source.

Reply:
Original source for 6th and 7th order rule: https://web.archive.org/web/20190609142211/http://people.sc.fsu.edu:80/~jburkardt/datasets/quadrature_rules_tet/quadrature_rules_tet.html

Reply:
8th order rule is something else, I'll try to verify.

Reply:
Seems to be collapsed tensor rule: https://github.com/FEniCS/fiat/blob/7d418fa0a372ac6f5e103533ab77ad6a9fac764c/FIAT/quadrature.py#L280

Reply:
I wonder if this 8th order Keast rule is better? https://web.archive.org/web/20120104033925/http://people.sc.fsu.edu/~jburkardt/datasets/quadrature_rules_tet/keast8_x.txt

Reply:
There seems to be much larger number of points in the rule used in this PR. Collapsed rules seem to be that way. Let us verify if it's better to use rules from https://web.archive.org/web/20120104033925/http://people.sc.fsu.edu/~jburkardt/datasets/quadrature_rules_tet/keast8_x.txt

Reply:
> Collapsed rules seem to be that way. Let us verify if it's better to use rules from web.archive.org/web/20120104033925/http://people.sc.fsu.edu/~jburkardt/datasets/quadrature_rules_tet/keast8_x.txt

Should I take a crack at this or leave it for the future? Do all the convergence tests pass without the 8th order rules?

Reply:
I decided to use the 8th order keast rule and it seems to pass the tests for convergence (at least locally for the moment). 

```
======================================================================================================================= test session starts =============================================================================================================================================================
platform win32 -- Python 3.9.2, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: D:\Research\scikitFEMContrib\scikit-fem\tests
plugins: anyio-2.2.0, hypothesis-6.13.7, cov-2.11.1, dependency-0.5.1
collected 25 items
                                                                                                                                                                                                                                                              [100%]
test_convergence.py .........................
                               [100%]                                                                                                                                                                                          =====================================

================================================================================================================== 25 passed in 98.99s (0:01:38) ===================================================================================================================

```

Reply:
It seems that some of these quadrature rules have negative weights: https://github.com/kinnala/scikit-fem/pull/653/files#diff-a32c8dc529b4ef2327e933b5e85b5ec80e8a108709027ef19e83ec7df8625fbbR574

It's not often an issue, but sometimes if you are performing minor variational crimes, say that your material field changes discontinuously inside the element and you are not meshing exactly the discontinuous interface, then this could cause non-positive-definite stiffness matrix.

However, we can leave it for now and fix it later if we notice any issues or find better high-order rules for tetrahedra.